src/clusterfuzz/_internal/metrics/monitor.py

```diff
@@ -114,13 +115,21 @@ def run(self):
                    f'metric_kind: {series.metric_kind}, '
                    f'start_time: {metric_st_sec}.{metric_st_ns}, '
                    f'end_time: {metric_et_sec}.{metric_et_ns}')
-          time_series.append(series)
+
+          if (series.metric_kind == metric_pb2.MetricDescriptor.MetricKind.GAUGE  # pylint: disable=no-member
+             ):
+            gauge_series.append(series)
+          else:
+            time_series.append(series)
 
           if len(time_series) == MAX_TIME_SERIES_PER_CALL:
+            time_series.extend(gauge_series)
```

Comment:
It seems to me we should check the combined length of `time_series` and `gauge_series` against `MAX_TIME_SERIES_PER_CALL` if we're to concatenate them. Otherwise we get this weird behavior where the max only applies to a subset of the entries in the list we send to the monitoring service, whose length is effectively unbounded.

Reply:
Done. Added sort based solution instead.